pkg/pillar/cmd/domainmgr/containerd.go

```diff
@@ -188,12 +272,210 @@ func getRktPodManifest(PodManifestFile string) (RktPodManifest, error) {
+	if err != nil {
+		return fmt.Errorf("Exception while writing container pid to %v/%v. %v", appDir, pidFilename, err)
+	}
+	_, err = executeShellCommands(mountCommands)
+	if err != nil {
+		return fmt.Errorf("Exception while writing mount rootfs. Error: %v", err)
+	}
+
+	return nil
+}
+
+func getBundleMountCommand(ctrdCtx context.Context, ctrdClient *containerd.Client, mountKey string, mountTarget string) ([]*command, error) {
+	snapshotter := ctrdClient.SnapshotService(defaultSnapshotter)
+	mounts, err := snapshotter.Mounts(ctrdCtx, mountKey)
+	if err != nil {
+		return nil, fmt.Errorf("Exception while fetching container snapshot mounts. %v", err)
+	}
+	return getMountCommands(mountTarget, mounts), nil
+}
+
+func getImageInfo(ctrdCtx context.Context, image containerd.Image) (v1.Image, error) {
+	var ociimage v1.Image
+	ic, err := image.Config(ctrdCtx)
+	if err != nil {
+		return ociimage, fmt.Errorf("getImageConfig: ubable to fetch image: %v config. %v", image.Name(), err.Error())
+	}
+	switch ic.MediaType {
+	case v1.MediaTypeImageConfig, images.MediaTypeDockerSchema2Config:
+		p, err := content.ReadBlob(ctrdCtx, image.ContentStore(), ic)
+		if err != nil {
+			return ociimage, fmt.Errorf("getImageConfig: ubable to read cotentStore of image: %v config. %v", image.Name(), err.Error())
+		}
+
+		if err := json.Unmarshal(p, &ociimage); err != nil {
+			return ociimage, fmt.Errorf("getImageConfig: ubable to marshal cotentStore of image: %v config. %v", image.Name(), err.Error())
+
+		}
+	default:
+		return ociimage, fmt.Errorf("unknown image config media type %s", ic.MediaType)
+	}
+	return ociimage, nil
+}
+
+func getImageInfoJSON(ctrdCtx context.Context, image containerd.Image) (string, error) {
+	ociimage, err := getImageInfo(ctrdCtx, image)
+	if err != nil {
+		return "", fmt.Errorf("getImageInfoJSON: ubable to fetch image: %v. %v", image.Name(), err.Error())
+	}
+	return getJSON(ociimage)
+}
+
+// Util methods
+
+func buildEnvVarsSlice(envMap map[string]string) []string {
+	var envVars []string
+
+	for envName, envValue := range envMap {
+		envVars = append(envVars, fmt.Sprintf("%s=%s", envName, envValue))
+	}
+
+	return envVars
+}
+
+// getJSON - returns input in JSON format
+func getJSON(x interface{}) (string, error) {
+	b, err := json.MarshalIndent(x, "", "    ")
+	if err != nil {
+		return "", fmt.Errorf("Exception while unmarshalling container spec JSON. %v", err)
+	}
+	return fmt.Sprint(string(b)), nil
+}
+
+// getMountCommands - returns list of Unix mount commands for target and list of mounts
+func getMountCommands(target string, mounts []mount.Mount) []*command {
+	var mountCommands []*command
+	for _, m := range mounts {
+		mountCommands = append(mountCommands, &command{
+			name: "mount",
+			args: []string{"-t", m.Type, m.Source, target, "-o", strings.Join(m.Options, ",")},
+		})
+	}
+	return mountCommands
+}
+
+func executeShellCommands(commands []*command) ([]string, error) {
```

Comment:
As above, we should be able to get rid of this if we use direct commands like [Mount](https://godoc.org/golang.org/x/sys/unix#Mount)